cmake/compiler/xcc.cmake

```diff
@@ -0,0 +1,7 @@
+include(${ZEPHYR_BASE}/cmake/compiler/gcc.cmake)
+
+set(NOSTDLIB_LDFLAGS
+  -nostdlib
+)
+
+set(NOSYSDEF_CFLAG "")
```

Comment:
Isnt' this dangerous?

Then on xcc it will be 

-D__SOMETHING_THAT_SHOULD_HAVE_BEEN_UNDEFINED

Reply:
My mistake, I thought -undef was a different flag ...